lib/rubygems/gem_runner.rb

```diff
@@ -24,6 +25,15 @@
 # classes they call directly.
 
 class Gem::GemRunner
+  include Gem::UserInteraction
+
+  HAVE_OPENSSL = begin
+    require 'openssl'
+    true
+  rescue LoadError
+    false
+  end
```

Comment:
You can instead add a toplevel `require 'rubygems/openssl'`, and then check `defined?(OpenSSL)` to keep things consistent with what we do in other places.

Reply:
It seems like the checks are not a fan of `defined?(OpenSSL)` 😭 . But maybe I'm doing this wrong?

Reply:
I believe you might need to rebase. The file `rubygems/openssl` I only introduced it very recently to unify all the `openssl` loading code. You shouldn't need to wrap the require with any `LoadError` rescuing because that's handled inside that file.